internal/agent/agent_test.go

```diff
@@ -113,6 +113,33 @@ var _ = Describe("Device Agent behavior", func() {
 			})
 		})
 
+		When("device/enrollmentrequests have been deleted", func() {
```

Comment:
I get the intent but I believe we have also already agreed that this flow is not stable or even desired. So why are we adding a test case?  We should have a clear path for this moving forward.

1.  proposed bug/feature
1.  will it be a supported flow
1.  design/implement

Reply:
Yes, I was blindly following the description of MGMT-17100 but it doesn't make much sense to support as a workflow. If key/csr are deleted, the device can't continue.

May be we should test for...

1) csr has been lost (a device looses power mid-enrollment) ... and but we guarantee that on the next attempt it will grab the CSR from the enrollment request.

2) device has csr+key, but the device does not exist anymore (has been deleted), what do we expect in that case, a new enrollment request to happen? (at least until we have more complex decomission flows).